### DISCUSSION TITLE
Documentation for the theory behind the Discrete galerkin operators in MFEM

### DISCUSSION THREAD

**Discussion by prathamgit** [mfem/mfem#2683] [Q&A]
I was trying to implement a robin-type boundary condition, but I quite couldn't figure out why it wasn't working out for me whereas it is working fine in ex27 of MFEM examples. I was wondering if there is any documentation explaining the equations that are implemented in various MFEM operators and functions, so that I could see exactly what it it that I am doing wrong.

**Comment by v-dobrev** [comment]
In Example 27, the Robin boundary conditions are imposed weakly by adding terms to both: (1) the bilinear form (the matrix):
https://github.com/mfem/mfem/blob/acb9cfa32196ed8f79c8779ff0cf8bc8d90f8485/examples/ex27.cpp#L215-L216 and (2) the linear form (the r.h.s.): https://github.com/mfem/mfem/blob/acb9cfa32196ed8f79c8779ff0cf8bc8d90f8485/examples/ex27.cpp#L245-L246

The coefficient `m_rbcACoef` is the product of the diffusivity constant `m` and the constant `a` from the Robin boundary condition:
```ascii
n.(m grad(u)) + (m a) u = m b
```
Similarly, `m_rbcBCoef` is the product of `m` and `b`. The `Array<int>` `rbc_bdr` describes on which part of the boundary (in terms of the boundary attributes) the Robin b.c. are imposed.
